core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -256,7 +265,9 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::MDRangePolicy<Traits...>,
             .exec_range();
       });
     });
-    q.ext_oneapi_submit_barrier(std::vector<sycl::event>{parallel_for_event});
+    // This barrier would prevent q.memcpy for subsequent kernels from being
+    // brought forward in time.
+    // q.ext_oneapi_submit_barrier(std::vector<sycl::event>{parallel_for_event});
```

Comment:
Should we just remove it then rather than comment it?  @masterleinad

Reply:
I've reinstated this barrier & removed the comment. I think @masterleinad is right that the barriers are necessary.